src/client/Microsoft.Identity.Client/Platforms/Features/WamBroker/WamBroker.cs

```diff
@@ -251,10 +241,26 @@ private static void AddAuthorityParamToRequest(AuthenticationRequestParameters a
                     throw new MsalClientException(MsalError.AuthenticationCanceledError, "WAM Account Picker did not return an account.");
                 }
 
+                bool isConsumerTenant = _webAccountProviderFactory.IsConsumerProvider(accountProvider);
                 // WAM returns the tenant here, not the full authority
-                bool isConsumerTenant = string.Equals(accountProvider.Authority, "consumers", StringComparison.OrdinalIgnoreCase);
                 wamPlugin = (isConsumerTenant && !isMsaPassthrough) ? _msaPlugin : _aadPlugin;
 
+                string transferToken = null;
+
+                if (isConsumerTenant && isMsaPassthrough)
+                {
+                    // Get a transfer token to avoid prompting the user twice
+                    transferToken = await _msaPassthroughHandler.FetchTransferTokenAsync(
+                       authenticationRequestParameters,
+                       accountProvider).ConfigureAwait(false);
+
+                    // For MSA-PT, the MSA provider will issue v1 token, which cannot be used.
+                    // Only the AAD provider can issue a v2 token
```

Comment:
this is for apps that were previously v1 apps, correct? and now the apps are migrating to add support for MSA, or converge an MSA registration w/an ADD registration...apps that are v2 will not have this issue. right?

Reply:
yes, v2 apps will not have this issue. However, there is no migration. Apps simply need to use a different client_id, which means they will force re-auth. This is preventing migration.